### PR TITLE
Ncnews11

### PR DESCRIPTION
**Pull request by Nat2llaw**
sort by and order queries

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -32,142 +32,263 @@ describe("GET/api/topics", () => {
...
+      .get("/api/articles?sort_by=author")
+      .expect(200)
+      .then(({ body: article }) => {
+        expect(article).toHaveLength(12);
+        expect(article).toBeSortedBy("author", {
+          descending: true,
+        });
+      });
+  });
+  test("200: return all the articles with authors in descending order", () => {
```
Comment (corkesw) [inline __tests__/app.test.js:92]: The description should say ascending

File: __tests__/app.test.js [hunk 2]
```diff
@@ -32,142 +32,263 @@ describe("GET/api/topics", () => {
...
+      });
+  });
+  test("404: return error for invalid query", () => {
+    return request(app)
+      .get("/api/articles?sort_by=banana")
+      .expect(404)
+      .then(({ body: article }) => {
+        expect(article.msg).toBe("Invalid query");
+      });
+  });
```
Comment (corkesw) [inline __tests__/app.test.js:121]: You should add an error test for an invalid order

File: db/model/model.js [hunk 3]
```diff
@@ -8,57 +8,129 @@ exports.fetchTopics = () => {
...
+      `INSERT INTO comments (article_id, author, body) VALUES ($1, $2, $3) RETURNING *`,
+      [id, newComment.username, newComment.body]
+    )
+    .then(({ rows: [comment] }) => {
+      return comment;
+    });
+};
+
+exports.fetchAllArticles = (topicQuery, sortByQuery, orderQuery) => {
+  if (topicQuery) {
```
Comment (corkesw) [inline db/model/model.js:63]: We should be able to combine the topic filter with the sort_by and order queries but we would be unable to do this with your current logic.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -32,142 +32,263 @@ describe("GET/api/topics", () => {
...
+    return request(app)
+      .get("/api/articles?order=asc")
+      .expect(200)
+      .then(({ body: article }) => {
+        expect(article).toHaveLength(12);
+        expect(article).toBeSortedBy("created_at", {
+          descending: false,
+        });
+      });
+  });
```
Comment (corkesw) [inline __tests__/app.test.js:113]: A 200 test that combines the topic filter with sort_by and order would be worthwhile here

**Review by corkesw — COMMENTED** [review]
Good work but there is a little more testing here. I would also like to see the code in the model being much DRYer. Please be careful with you commits and branching as there is a lot of committed code here that is just not relevant to the ticket.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -32,142 +32,263 @@ describe("GET/api/topics", () => {
...
+});
 
 describe("GET/api/articles", () => {
-  test("200: return all the articles sort by date descending", () => {
+  test("200: return all the articles sorted by date descending by default", () => {
     return request(app)
       .get("/api/articles")
       .expect(200)
       .then(({ body: article }) => {
-        expect(article).toHaveLength(25);
```
Comment (jimstev2447) [inline __tests__/app.test.js:51]: why have you removed this part of the test, its still valid

File: __tests__/app.test.js [hunk 6]
```diff
@@ -32,142 +32,263 @@ describe("GET/api/topics", () => {
...
-              title: expect.any(String),
-              topic: expect.any(String),
-              author: expect.any(String),
-              body: expect.any(String),
-              created_at: expect.any(String),
-              votes: expect.any(Number),
-              comment_count: expect.any(Number),
-            })
-          );
+        expect(article).toHaveLength(11);
```
Comment (jimstev2447) [inline __tests__/app.test.js:64]: its fine here though because we are just looking at the query

File: __tests__/app.test.js [hunk 7]
```diff
@@ -32,142 +32,263 @@ describe("GET/api/topics", () => {
...
-          .get("/api/articles/banana")
+          .patch("/api/articles/1")
+          .send({ inc_votes: "banana" })
           .expect(400)
           .then(({ body }) => {
             expect(body.msg).toBe("Bad Request");
           });
       });
     });
+    test("400: article id not in database", () => {
```
Comment (jimstev2447) [inline __tests__/app.test.js:195]: I dont really understand why all these changes are here, it feels like you're adding code that's not in the ticket somehow?

**Pull request merged**